robyn/__init__.py

```diff
@@ -3,9 +3,13 @@
 import os
 import socket
 from pathlib import Path
-from typing import Callable, List, Optional, Tuple, Union
+from typing import Callable, List, Optional, Tuple, Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import multiprocessing as mp
+else:
+    import multiprocess as mp
```

Comment:
why are we importing from two different libraries?

Reply:
multiprocess is a direct replacement for multiprocessing however multiprocess has not added support for type checks by mypy etc (see https://github.com/uqfoundation/multiprocess/issues/128 ). As multiprocess is a direct drop in replacement we can use the types from multiprocessing by importing it only for type checking.

In the last commenton that multiprocess issue is a workaround. Import the type definitions from multiprocessing ONLY while in type-checking mode. see https://github.com/uqfoundation/multiprocess/issues/128#issuecomment-2188208560

Reply:
I've committed a comment to explain this workaround. Wondering if we should include a comment in pyproject.toml to check if this can be changed whenever we upgrade the dependency on multiprocess

Reply:
> multiprocess is a fork of multiprocessing. multiprocess extends multiprocessing to provide enhanced serialization, using dill .

Hey @dave42w 👋 

We use multiprocess for better serialization. Let's us not use two different libraries just because of type checking. We can disable mypy on the particular line.

Reply:
Oop's only just seen this. Will make that change.

Reply:
Done